lib/sdn_epc/converter.ex

```diff
@@ -0,0 +1,107 @@
+defmodule SdnEpc.Converter do
+  require SdnEpc.OfpmRecord
+
+  @ofp_version Application.get_env(:sdn_epc, :ofp_version)
+
+  def convert({:packet_in, xid, body}) do
```

Comment:
Almost there :) You're still duplicating this piece of code:

```elixir
SdnEpc.OfpmRecord.ofp_message(
       version: @ofp_version,
       xid: xid,
       body: packet_in_rec)
```

There could be another function, like `body/X` with many clauses matching on a particular message type.